src/modules/hip/hip_fused_functions.cpp

```diff
@@ -48,12 +46,539 @@ color_twist_hip_batch(Rpp8u* srcPtr, Rpp8u* dstPtr, rpp::Handle& handle, RppiChn
+
+#if defined (HIPRTC)
+
+    std::string kernel_file = "color_twist.cpp";
+    std::string kernel_name = "color_twist_batch";
+    get_kernel_name(kernel_name, tensor_info);
+
+    std::vector<size_t> vld{16, 16, 1};
+    std::vector<size_t> vgd{max_width, max_height, handle.GetBatchSize()};
+
+    handle.AddKernel("", "",  kernel_file, kernel_name, vld, vgd, "")(srcPtr,
+                                                                      dstPtr,
+                                                                      handle.GetInitHandle()->mem.mgpu.floatArr[0].floatmem,
+                                                                      handle.GetInitHandle()->mem.mgpu.floatArr[1].floatmem,
+                                                                      handle.GetInitHandle()->mem.mgpu.floatArr[2].floatmem,
+                                                                      handle.GetInitHandle()->mem.mgpu.floatArr[3].floatmem,
+                                                                      handle.GetInitHandle()->mem.mgpu.roiPoints.x,
+                                                                      handle.GetInitHandle()->mem.mgpu.roiPoints.roiWidth,
+                                                                      handle.GetInitHandle()->mem.mgpu.roiPoints.y,
+                                                                      handle.GetInitHandle()->mem.mgpu.roiPoints.roiHeight,
+                                                                      handle.GetInitHandle()->mem.mgpu.srcSize.height,
+                                                                      handle.GetInitHandle()->mem.mgpu.srcSize.width,
+                                                                      handle.GetInitHandle()->mem.mgpu.maxSrcSize.width,
+                                                                      handle.GetInitHandle()->mem.mgpu.srcBatchIndex,
+                                                                      handle.GetInitHandle()->mem.mgpu.inc,
+                                                                      handle.GetInitHandle()->mem.mgpu.dstInc,
+                                                                      in_plnpkdind,
+                                                                      out_plnpkdind);
+
+#elif defined(STATIC)
+
+    hip_exec_color_twist_batch(srcPtr, dstPtr, handle, in_plnpkdind, out_plnpkdind, max_height, max_width);
+
+#endif
+
+    return RPP_SUCCESS;
+}
+
+RppStatus
+color_twist_hip_batch_tensor_fp16(Rpp16f *srcPtr, Rpp16f *dstPtr, rpp::Handle &handle, RPPTensorFunctionMetaData &tensor_info)
+{
+    int in_plnpkdind = getplnpkdind(tensor_info._in_format);
+    int out_plnpkdind = getplnpkdind(tensor_info._out_format);
+    Rpp32u max_height, max_width;
+    max_size(handle.GetInitHandle()->mem.mgpu.csrcSize.height, handle.GetInitHandle()->mem.mgpu.csrcSize.width, handle.GetBatchSize(), &max_height, &max_width);
+
+#if defined (HIPRTC)
+
```

Comment:
It is better to remove empty code block

Reply:
As soon as this set of PRs go through, we'll be removing RTC entirely, so these HIPRTC blocks containing / not containing code will be deleted and only the hip_exec_* call below will be retained. (without the compile time switch)